hack/aws-acceptance-test-cleanup/main.go

```diff
@@ -759,6 +764,111 @@ func destroyBackoff(ctx context.Context, resourceKind string, resourceID string,
 	}, backoff.WithContext(expoBackoff, ctx))
 }
 
+func cleanupIAMRoles(ctx context.Context, iamClient *iam.IAM) error {
+	var rolesToDelete []*iam.Role
+	rolePrefix := "consul-k8s-"
+	err := iamClient.ListRolesPagesWithContext(ctx, &iam.ListRolesInput{},
+		func(page *iam.ListRolesOutput, lastPage bool) bool {
+			return filterRolesWithPrefix(page, lastPage, &rolesToDelete, rolePrefix)
+		})
+	if err != nil {
+		return fmt.Errorf("failed to list roles: %v", err)
+	}
```

Comment:
#nit I think this and the `filterRolesWithPrefix` func get a little easier to understand with less dereferencing and whatnot if we just let the func that we're calling out to collect the roles itself.

```go
rolesToDelete, err := filterIAMRolesWithPrefix(ctx, iamClient, "consul-k8s-")
if err != nil {
    return fmt.Errorf("failed to list roles: %w", err)
}
```

```go
func filterIAMRolesWithPrefix(ctx context.Context, iamClient *iam.IAM, prefix string) ([]*iam.Role, error) {
	var roles []*iam.Role

	err := iamClient.ListRolesPagesWithContext(ctx, &iam.ListRolesInput{},
		func(page *iam.ListRolesOutput, lastPage bool) bool {
			for _, role := range page.Roles {
				name := aws.StringValue(role.RoleName)
				if strings.HasPrefix(name, prefix) {
					roles = append(roles, role)
				}
			}

			// can just always return true here since we always want to see all of the roles
			return true
		})
	if err != nil {
		return nil, err
	}

	return roles, nil
}
```